query-engine/core/src/telemetry/models.rs

```diff
@@ -39,6 +55,11 @@ impl TraceSpan {
 
 impl From<SpanData> for TraceSpan {
     fn from(span: SpanData) -> Self {
+        let otel_kind = match span.attributes.get(&Key::from_static_str("otel.kind")) {
+            Some(Value::String(Cow::Borrowed("client"))) => OtelKind::Client,
+            _ => OtelKind::Internal,
+        };
```

Comment:
You must use `span.span_kind` here, `otel.kind` doesn't exist as an attribute anymore on the OTEL span: https://github.com/tokio-rs/tracing-opentelemetry/blob/v0.1.x/src/layer.rs#L442